### PR TITLE
fix: issue in switch, input, select

### PR DESCRIPTION
**Pull request by albinAppsmith**
## Description

This PR contains,
1. Added gap in switch label and component.
2. Added story types for switch onChange.
3. Added validationState for inputs and Select

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

- Manual on storybook 
- Manual on main repo
- Jest
- Cypress

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: packages/design-system/src/Input/Input.tsx [hunk 1]
```diff
@@ -46,6 +46,7 @@ const Input = forwardRef<HTMLInputElement, InputProps>(
       type = "text",
       UNSAFE_height,
       UNSAFE_width,
+      validationState,
```
Comment (tanvibhakta) [inline packages/design-system/src/Input/Input.tsx:49]: isValid?

File: packages/design-system/src/NumberInput/NumberInput.stories.tsx [hunk 2]
```diff
@@ -25,4 +25,5 @@ NumberInputStory.storyName = "NumberInput";
 NumberInputStory.args = {
   prefix: "$",
   suffix: "",
+  validationState: "valid",
```
Comment (tanvibhakta) [inline packages/design-system/src/NumberInput/NumberInput.stories.tsx:28]: isValid

File: packages/design-system/src/__config__/types.ts [hunk 3]
```diff
@@ -2,3 +2,5 @@
 export type Sizes = "xs" | "sm" | "md" | "lg" | "xl" | "xxl" | "xxxl" | "xxxxl";
 
 export type Kinds = "success" | "error" | "warning" | "info" | undefined;
+
+export type ValidationState = "valid" | "invalid";
```
Comment (tanvibhakta) [inline packages/design-system/src/__config__/types.ts:6]: This can just be a boolean.

**Review by tanvibhakta — CHANGES_REQUESTED** [review]

File: packages/design-system/src/Input/Input.tsx (hunk 1 above)
Comment (albinAppsmith) [reply packages/design-system/src/Input/Input.tsx:49]: This is provided by react-aria

Comment (tanvibhakta) [reply packages/design-system/src/Input/Input.tsx]: We can always expose the variable that matches our api pattern and mutate it ourselves to give react-aria the value it needs. I've made the change and approved the PR, feel free to merge.

**Comment by tanvibhakta** [comment]
If we're exporting all types, let's make the change in the generator file as well.

**Comment by albinAppsmith** [comment]
> If we're exporting all types, let's make the change in the generator file as well.

Updated

**Review by tanvibhakta — APPROVED** [review]

**Pull request merged**